python/tests/test_highlevel.py

```diff
@@ -885,6 +885,54 @@ def test_virtual_root_arg(self):
         assert self.t.mrca(0, 5) == 5
 
 
+class TestPathLength:
+    t = tskit.Tree.generate_balanced(9)
+    #         16
+    #    ┏━━━━┻━━━┓
+    #    ┃       15
+    #    ┃     ┏━━┻━┓
+    #   11     ┃   14
+    #  ┏━┻━┓   ┃  ┏━┻┓
+    #  9  10  12  ┃ 13
+    # ┏┻┓ ┏┻┓ ┏┻┓ ┃ ┏┻┓
+    # 0 1 2 3 4 5 6 7 8
+
+    def test_tmrca_leaf(self):
+        assert self.t.path_length(0, 16) == 3
+        assert self.t.path_length(16, 0) == 3
+        assert self.t.path_length(7, 16) == 4
+
+    def test_equal_depth(self):
+        assert self.t.path_length(5, 16) == self.t.depth(5)
+
+    def test_two_leaves(self):
+        assert self.t.path_length(0, 8) == 7
+
+    def test_two_leaves_depth(self):
+        assert self.t.path_length(0, 8) == self.t.depth(0) + self.t.depth(8)
+
+    def test_less_than_two_args(self):
+        with pytest.raises(ValueError):
+            self.t.path_length(1)
+
+    def test_no_args(self):
+        with pytest.raises(ValueError):
+            self.t.path_length()
+
+    def test_same_args(self):
+        assert self.t.path_length(10, 10) == 0
+
+    def test_different_tree_levels(self):
+        assert self.t.path_length(1, 10) == 3
+
+    def test_out_of_bounds_args(self):
+        with pytest.raises(ValueError):
+            self.t.path_length(0, 20)
+
+    # def test_virtual_root_arg(self):
```

Comment:
So, we need to think about the semantics of the virtual root here. I guess the simplest definition would be that 
```
tree.path_length(u, tree.virtual_root) == tree.depth(u) + 1
tree.path_length(tree.virtual_root, u) == tree.depth(u) + 1
tree.path_length(tree.virtual_root, tree.virtual_root) == 0
```
which, it turns out happens automatically :hooray:

Here's some tests for this and checking the no-mrca case:
```python
    @pytest.mark.parametrize("u", range(17))
    def test_virtual_root_arg(self, u):
        self.t.path_length(u, self.t.virtual_root) == self.t.depth(u)
        self.t.path_length(self.t.virtual_root, u) == self.t.depth(u)

    def test_both_args_virtual_root(self):
        self.t.path_length(self.t.virtual_root, self.t.virtual_root) == 0

    def test_no_mrca(self):
        tree = self.t.copy()
        tree.clear()
        assert math.isinf(tree.path_length(0, 1))

Reply:
Nice!

I was just wondering why I can't access `virtual_root` in this simple code:

```python
t = tskit.Tree.generate_balanced(9)
print(t.virtual_root)

> AttributeError: 'Tree' object has no attribute 'virtual_root'
```